contracts/token/ERC20/extensions/ERC20Votes.sol

```diff
@@ -27,10 +27,10 @@ abstract contract ERC20Votes is ERC20, Votes {
     error ERC20ExceededSafeSupply(uint256 increasedSupply, uint256 cap);
 
     /**
-     * @dev Maximum token supply. Defaults to `type(uint224).max` (2^224^ - 1).
+     * @dev Maximum token supply. Defaults to `type(uint208).max` (2^208^ - 1).
      */
-    function _maxSupply() internal view virtual returns (uint224) {
-        return type(uint224).max;
+    function _maxSupply() internal view virtual returns (uint256) {
```

Comment:
Why change this to `uint256`? Using the smaller type prevents returning a number that is too large for the data structure to support it.

```suggestion
    function _maxSupply() internal view virtual returns (uint208) {
```

Reply:
`uint208` makes it a type error to override with a function that returns `type(uint256).max`, for example, or any literal that is too large to fit uint208.

Reply:
What is multiple module implement that function with different values (votes limit to 208, but another module would limit to 192 or 128 for some other reason) how would we resolve that ?

If the return types are different it's a mess. If they are both uint256 there may be some hope

Reply:
This is not a concrete problem we have currently so I would go with the option that is safer.

Reply:
If that issue happens during the 5.x cycle, would we be able to change it without a major change ?

I don't think overriding this with a bigger function is something anyone would realistically do.

Reply:
Also the only place we use it, we do an implicit upcast to uint256

Reply:
If something like this happens in a minor release I think we would just add an internal function with a name other than `_maxSupply` so they wouldn't clash, and `ERC20Votes` could implement that internal function in terms of the existing one.

Reply:
If you feel strongly about this I can accept it. The risk if >uint208 is returned is simply that some transfers might revert but it's a recoverable situation.